django/db/models/sql/query.py

```diff
@@ -1244,16 +1245,16 @@ def build_filter(self, filter_expr, branch_negated=False, current_negated=False,
             raise FieldError("Cannot parse keyword query %r" % arg)
         lookups, parts, reffed_expression = self.solve_lookup_type(arg)
 
-        self.check_filterable(reffed_expression)
```

Comment:
Moving this line is not related with a bugfix. Please revert it and add
```python
if check_filterable:
    ...
```
in both places.

Reply:
Done